frontend/src/components/history/attemptInfo.js

```diff
@@ -0,0 +1,145 @@
+import '../../App.css';
+import React from 'react';
+
+import Button from '@mui/material/Button';
+import Chip from '@mui/material/Chip';
+import { styled } from '@mui/material/styles';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Typography from '@mui/material/Typography';
+
+import ComplexityChip from '../questions/complexityChip';
+
+const BootstrapDialog = styled(Dialog)(({ theme }) => ({
+  '& .MuiDialogContent-root': {
+    padding: theme.spacing(2),
+  },
+  '& .MuiDialogActions-root': {
+    padding: theme.spacing(1),
+  },
+}));
+
+function AttemptInfo({ open, handleClose, question, attempt }) {
+
+  // if there is no selected question, return null
+  if (question === undefined) {
+    return null;
+  }
+
+  const renderedParts = renderDescription(question.description);
+  const renderedAttempt = attempt;
+
+  return (
+    <div>
+      <BootstrapDialog
+        onClose={handleClose}
+        aria-labelledby="customized-dialog-title"
+        open={open}
+        fullWidth
+      >
+        {/* Question title */}
+        <DialogTitle sx={{ m: 0, p: 2 }} id="customized-dialog-title">
+          {question.title}
+        </DialogTitle>
+
+        {/* Rest of question contents in dialog box */}
+        <DialogContent dividers>
+          <Typography gutterBottom>
+            <b>ID:</b> {question.id}
+          </Typography>
+          <Typography gutterBottom sx={{ wordBreak: "break-word" }} component={'span'}>
+            <b>Category:</b> {question.categories.map((category) => (
+              <Chip key={category} label={category}></Chip>
+            ))}
+          </Typography>
+          <br />
+          <Typography gutterBottom sx={{ wordBreak: "break-word" }} component={'span'}>
+            <b>Complexity:</b> <ComplexityChip complexity={question.complexity} />
+          </Typography>
+          <Typography gutterBottom sx={{ whiteSpace: "pre-line" }} component="div">
+            {renderedParts}
+          </Typography>
+            <b>Your Attempt:
+            </b>
+            <Typography gutterBottom sx={{ whiteSpace: "pre-line" }} component="div">
+              {renderedAttempt}
+            </Typography>
+        </DialogContent>
+
+        {/* Action buttons in bottom right of the dialog */}
+        <DialogActions>
+          <Button onClick={handleClose}>
+            Close
+          </Button>
+        </DialogActions>
+      </BootstrapDialog>
+    </div>
+  );
+}
+
+function renderDescription(text) {
```

Comment:
fyi this function is removed from questionInfo, you can remove it or leave it but i feel like it doesn't really work with Leetcode's question format